tsdb/engine/tsm1/encoding.gen.go.tmpl

```diff
@@ -53,18 +53,23 @@ func (a {{.Name}}Values) assertOrdered() {
 // Deduplicate returns a new slice with any values that have the same timestamp removed.
 // The Value that appears last in the slice is the one that is kept.
 func (a {{.Name}}Values) Deduplicate() {{.Name}}Values {
-	m := make(map[int64]{{.Name}}Value)
-	for _, val := range a {
-		m[val.UnixNano()] = val
+	if len(a) == 0 {
+		return a
 	}
+	sort.Stable(a)
+	var i int
+	for j, v := range a {
```

Comment:
How about for j := 1; j < len(a); j++{ and then you can drop the if j == 0 check?